### PR TITLE
Add paypal to Weekly, Weekly gift and Paper checkouts

### PR DESCRIPTION
**Pull request by lucymonie**
## Why are you doing this?
So that customers can pay using Paypal in all checkouts.

[**Trello Card**](https://trello.com/c/hIIcn9y7/2707-add-paypal-to-other-checkouts)

## Changes
* Add Paypal to Guardian Weekly checkout
* Add Paypal to Guardian Weekly gift checkout
* Add Paypal to Paper checkout
* Refactor Paypal and Direct Debit submit buttons back into the paymentSelector

## Screenshots
Screen Shot 2019-12-18 at 13 01 30

Screen Shot 2019-12-18 at 13 01 52

Screen Shot 2019-12-18 at 13 02 16

### CODE REVIEW

File: support-frontend/assets/components/paypalExpressButton/PayPalExpressButton.jsx [hunk 1]
```diff
@@ -55,7 +55,7 @@ export class PayPalExpressButton extends React.Component<PropTypes> {
     if (!this.props.hasLoaded) {
       return null;
     }
-
+    console.log('hasLoaded: ', this.props.hasLoaded);
```
Comment (alexflav23) [inline support-frontend/assets/components/paypalExpressButton/PayPalExpressButton.jsx]: Do we want to leave this behind?

File: support-frontend/assets/pages/paper-subscription-checkout/components/paperCheckoutForm.jsx [hunk 2]
```diff
@@ -103,18 +112,29 @@ function mapStateToProps(state: WithDeliveryCheckoutState) {
     isTestUser: state.page.checkout.isTestUser,
     country: state.common.internationalisation.countryId,
     stripeSetupIntentEndpoint: getGlobal('stripeSetupIntentEndpoint'),
+    csrf: state.page.csrf,
+    currencyId: state.common.internationalisation.currencyId,
+    payPalHasLoaded: state.page.checkout.payPalHasLoaded,
+    amount: getProductPrice(
```
Comment (alexflav23) [inline support-frontend/assets/pages/paper-subscription-checkout/components/paperCheckoutForm.jsx]: Looks like we could pass in a checkout object to this method, unless it's used in other places. Not really important for this PR though.

File: support-frontend/assets/pages/digital-subscription-checkout/components/digitalCheckoutForm.jsx [hunk 3]
```diff
@@ -160,6 +158,8 @@ function DigitalCheckoutForm(props: PropTypes) {
   const PriceSummary = () =>
     <p className="component-credit-card-price">{priceSummary}</p>;
 
+  console.log('payPalHasLoaded: ', props.payPalHasLoaded);
```
Comment (rupertbates) [inline support-frontend/assets/pages/digital-subscription-checkout/components/digitalCheckoutForm.jsx]: Could remove this now

File: support-frontend/assets/components/paypalExpressButton/PayPalExpressButton.jsx (hunk 1 above)
Comment (lucymonie) [reply support-frontend/assets/components/paypalExpressButton/PayPalExpressButton.jsx]: Yes! 😳

File: support-frontend/assets/pages/digital-subscription-checkout/components/digitalCheckoutForm.jsx (hunk 3 above)
Comment (lucymonie) [reply support-frontend/assets/pages/digital-subscription-checkout/components/digitalCheckoutForm.jsx]: Wow, two console logs... 😬

File: support-frontend/assets/pages/paper-subscription-checkout/components/paperCheckoutForm.jsx (hunk 2 above)
Comment (lucymonie) [reply support-frontend/assets/pages/paper-subscription-checkout/components/paperCheckoutForm.jsx]: That would be nice, however it has some optional args, so I might leave it for the moment.

**Pull request merged**